### PR TITLE
feature: Add diff coverage quality gate to Quality Settings CY-5568

### PR DESCRIPTION
**Pull request by prcr**
Adds the new diff coverage quality gate rule to the Quality Settings page, explicitly mentioning that this rule only exists for pull requests.

### :eyes: Live preview
https://deploy-preview-1044--docs-codacy.netlify.app/repositories-configure/adjusting-quality-settings/#gates

### :construction: To do
- [x] Update screenshots
- [x] Wait for [CY-5534](https://codacy.atlassian.net/browse/CY-5534) to be released

### CODE REVIEW

**Review by inesgomas — APPROVED** [review]

**Comment by vascorsd** [comment]
This live preview doesn't match up :sweat: - https://docs.codacy.com/faq/general/how-do-i-block-merging-prs-using-codacy-as-a-quality-gate/#configuring-gate with the PR.

Should be https://deploy-preview-1044--docs-codacy.netlify.app/faq/general/how-do-i-block-merging-prs-using-codacy-as-a-quality-gate/#configuring-gate :)

**Review by vascorsd — APPROVED** [review]

**Comment by machadoit** [comment]
@prcr Is there a place where we could sneak some diff coverage definition? Like on that page, or a "know more about Diff coverage"? 🤔 

Something like:
* Diff coverage is the percentage of new or modified lines that are covered by tests

(Shameless copy paste from https://diff-cover.readthedocs.io/en/latest/README.html :D)

File: docs/repositories-configure/adjusting-quality-settings.md [hunk 1]
```diff
@@ -21,19 +23,19 @@ These settings configure when Codacy reports pull requests and commits as not up
 -   **New security issues are over:** Pull requests or commits are marked not up to standards if the number of security issues introduced is higher than the set value. By default, this setting is off.
 -   **Complexity is over:** Pull requests or commits are marked not up to standards if the introduced complexity is higher than the set value. By default, this setting is off.
 -   **Duplication is over:** Pull requests or commits are marked not up to standards if the number of clones introduced is higher than the set value. By default, this setting is off.
+-   **Diff coverage is under:** Pull requests are marked not up to standards if the code coverage of the lines added or changed by the pull request is below the set value.
```
[the remark this answers is not in the document]
Comment (prcr) [reply docs/repositories-configure/adjusting-quality-settings.md:26]: I understand what you mean, and indeed it makes more sense to have the definition in a centralized place. :thinking: 

Currently, we have the definition of diff coverage [on the pull requests page](https://docs.codacy.com/repositories/pull-requests/#pull-request-quality-overview) (adapted from the definition that you mention, but subject to improvements :slightly_smiling_face:). On the other hand, I have plans for a page where we list and clearly define **all metrics** that Codacy supports, including the new diff coverage: [DOCS-25](https://codacy.atlassian.net/browse/DOCS-25).

The best idea that I have to solve this in the short term is to include a link to the commit/pull request quality overview section for a list of all the metrics that the quality gate rules are based on. But I'll let this sit until tomorrow and perhaps I'll think of something better. :bulb:

[the remark this answers is not in the document]
Comment (prcr) [reply docs/repositories-configure/adjusting-quality-settings.md]: @machadoit Please have a look at this new iteration:

https://deploy-preview-1044--docs-codacy.netlify.app/repositories-configure/adjusting-quality-settings/#gates

Now, the introduction of the **Gates** section explains that the rules affect the color of the metrics on the pull request or commit pages (where users are able to learn more about each metric). And indeed the explanation for the diff coverage rule becomes more straightforward. :+1:

**Review by pedrocodacy — APPROVED** [review]

File: docs/repositories-configure/adjusting-quality-settings.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (machadoit) [reply docs/repositories-configure/adjusting-quality-settings.md]: Nice, yeah it looks good! I like to read the explanation by itself.

**Review by machadoit — APPROVED** [review]

**Pull request merged**